### PR TITLE
[20568] Remove all redundant headers from include/fastrtps

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR removes all redundant headers from include/fastrtps. 
It makes UnitsParser private 
It depends on:

* #4545 

Related PRs:

* eProsima/Discovery-Server#73
* eProsima/ShapesDemo#133

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- _N/A_: Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_: Any new/modified methods have been properly documented using Doxygen. 
- ❌ Changes are ABI compatible. 
- ❌ Changes are API compatible. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 
    Related documentation PR: eProsima/Fast-DDS-docs#737
- _N/A_: Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/utils/UnitsParser.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@
 #include <regex>
 #include <stdexcept>
 
-#include <fastrtps/utils/UnitsParser.hpp>
+#include <utils/UnitsParser.hpp>
```
Comment (elianalf) [inline src/cpp/utils/UnitsParser.cpp:28]: I think we are following the google style even if the headers are private, so this should be on top of the others

**Review by elianalf — CHANGES_REQUESTED** [review]

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Review by elianalf — DISMISSED** [review]

**Comment by EduPonz** [comment]
ASan workflow run with correct DS branch [here](https://github.com/eProsima/Fast-DDS/actions/runs/8464544701)

**Comment by elianalf** [comment]
We should create a Shapes-Demo branch with these changes in particular the building failed because it couldn't find `fastrtps/rtps/common/Guid.h`

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Review by elianalf — APPROVED** [review]

**Review by elianalf — APPROVED** [review]

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Pull request merged**